Java client: autogenerated bindings

batiati opened the pull request:
This PR adds autogenerated Java bindings: 

- `src/clients/java/java_bindings.zig`: This is the most complex code generator we have until now. Due to Java's nature, we need to generate each class into its own file, add javadocs metadata and take care of things such as get/set properties and visibility modifiers.
Any suggestions to make it clearer are welcome.

- Zig tests asserting that the current Java bindings present on the repository are up to date.

- Java tests for new elements introduced (constants that now are rendered as enums and other minor changes).

Related: #298

batiati committed:
Java client: autogenerated bindings

batiati committed:
Java client: Tests passing

batiati committed:
Java client: More test coverage

- Minor cleanup.
- Test coverage at 97%, including the generated code.

batiati committed:
Java client: Remove line wraps from autogenerated comments.

sentientwaffle commented on src/clients/java/src/main/java/com/tigerbeetle/AccountFlags.java:
```diff
@@ -1,10 +1,27 @@
+    /**
+     * @see <a href="https://docs.tigerbeetle.com/reference/accounts#flagslinked">linked</a>
+     */
+    int LINKED = (int) (1 << 0);
```
Why are these now `int` instead of `short`?

batiati replied:
For better API ergonomy, we expose 16-bit unsigned integers in Java as `int` instead of `short`.
Even though, they are always validated and stored with the correct type.

sentientwaffle commented on src/clients/java/src/test/java/com/tigerbeetle/AccountTest.java:
```diff
@@ -147,6 +148,37 @@ public void testCodeOverflow() {
         accounts.setCode(Integer.MAX_VALUE);
     }
 
+    public void testReserved() {
```
Why do some of these methods have `@Test` and others don't?

batiati replied:
Thank you for catching that ... I found a few others in the code, it's hard to spot!

batiati committed:
Java client: autogenerated bindings review

- Fixed missing @Test attributes on Java tests.

- Better comments and line wraps on java_bindings.zig

batiati committed:
Merge branch 'main' into batiati-java-bindings

batiati committed:
Zig fmt
